Dockerfile-oss

```diff
@@ -0,0 +1,115 @@
+FROM --platform=$BUILDPLATFORM maven:3.6.3-jdk-8-slim as builder
```

Comment:
When using buildx, there isn't a way to make a build of an image in one invocation of buildx available to a subsequent invocation of buildx without the use of a registry. This means our traditional approach of having a 'build' docker image that we then reference in subsequent docker files will not work (unless we push it to a registry). To get around this, we use one multi-stage docker file that contains the build stages. A further discussion of this issue in buildx can be found here: https://github.com/docker/buildx/issues/156

Reply:
Also, note the `--platform=$BUILDPLATFORM`. This is because we don't have to build the jar for each arch we support since its java. This indicates that this stage should only be built on whatever the build platform is (probably linux-amd64).